src/lib/reasoners/shostak.ml

```diff
@@ -369,7 +386,14 @@ struct
     | Enum t -> ENUM.is_constant t
     | Adt t -> ADT.is_constant t
     | Ite t -> ITE.is_constant t
-    | Ac _ | Term _ -> false
+    | Term t ->
+      begin
+        let Expr.{ f; xs; _ } = Expr.term_view t in
+        match f, xs with
+        | Symbols.(True | False | Void), [] -> true
+        | _ -> false
+      end
```

Comment:
There should be a comment here explaining why this is not `Expr.is_const_term` (because other constants are semantic values).

Reply:
In fact, we could use `Expr.is_const_term`. I didn't use it because when I wrote this code, the predicate `Expr.is_const_term` should accept variables as we use it in the smart constructor of let bindings. I decided to use two predicates and we don't need to accept variables. Still, it makes sense to write a specific code here. I add a comment to explain the situation.